### PR TITLE
[Flow test] Resolve base64 in flow details to image path

### PR DESCRIPTION
**Pull request by lalala123123**
# Description
bug: streamlit will use base64 to record image in the chat history. 
image

Expect:
Image path is shown in test details.

How to fix:
Convert all base64 multi media in test details to path

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/_constants.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 FLOW_TOOLS_JSON = "flow.tools.json"
 FLOW_TOOLS_JSON_GEN_TIMEOUT = 60
 PROMPT_FLOW_DIR_NAME = ".promptflow"
+PROMPT_FLOW_INTERMEDIATE_NAME = "intermediate"
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_constants.py:21]: PROMPT_FLOW_INTERMEDIATE_NAME  -> PROMPT_FLOW_INTERMEDIATE_**DIR**_NAME ?

File: src/promptflow/promptflow/_sdk/_utils.py [hunk 2]
```diff
@@ -897,3 +904,34 @@ def dump_flow_result(flow_folder, prefix, flow_result=None, node_result=None):
     if output:
         with open(dump_folder / f"{prefix}.output.json", "w") as f:
             json.dump(output, f, indent=2)
+
+
+def _resolve_base64_image_in_test_result(test_result, flow_folder, sub_dir):
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_utils.py:909]: Is this PR a ask from VSC?

File: src/promptflow/promptflow/_sdk/_utils.py [hunk 3]
```diff
@@ -897,3 +904,34 @@ def dump_flow_result(flow_folder, prefix, flow_result=None, node_result=None):
...
+    """
+    Resolve base64 to image path in the test result.
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_utils.py:911]: _resolve sounds like reading from file to memory obj. But you are doing the opposite here. Maybe name this as: 

_save_base64_image_to_file()

File: src/promptflow/promptflow/_sdk/_constants.py (hunk 1 above)
Comment (lalala123123) [reply src/promptflow/promptflow/_sdk/_constants.py]: update constant name

File: src/promptflow/promptflow/_sdk/_utils.py (hunk 2 above)
Comment (lalala123123) [reply src/promptflow/promptflow/_sdk/_utils.py]: Yes, this ark is related to VSC, reported by Heyi. After changing to the path format, customer can jump directly to the image file in VSC
image

**Review by wangchao1230 — DISMISSED** [review]

**Review by YingChen1996 — DISMISSED** [review]

**Review by crazygao — APPROVED** [review]

**Review by YingChen1996 — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**